shunting-yard-common/src/test/java/com/hotels/shunting/yard/common/messaging/MessageTest.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (C) 2016-2019 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.shunting.yard.common.messaging;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import java.lang.reflect.InvocationTargetException;
+
+public class MessageTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void checkNull() throws InvocationTargetException {
+    final Message.Builder objectUnderTest = Message.builder();
+    objectUnderTest.database("test_db");
+    objectUnderTest.table("test_table");
+    objectUnderTest.payload(null);
+    objectUnderTest.timestamp(1515585600000L);
+
+    assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(() -> objectUnderTest.build())
+        .withMessage("Parameter 'payload' is required");
+  }
+
+  @Test
+  public void checkEmpty() throws InvocationTargetException {
+    final Message.Builder objectUnderTest = Message.builder();
+    objectUnderTest.database("test_db");
+    objectUnderTest.table("");
+    objectUnderTest.payload("foo");
+    objectUnderTest.timestamp(1515585600000L);
+
+    assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(() -> objectUnderTest.build())
+        .withMessage("Parameter 'table' is required");
+  }
+
+  @Test
+  public void buildOutputNotNull() throws InvocationTargetException {
+    final Message.Builder objectUnderTest = Message.builder();
+    objectUnderTest.database("test_db");
+    objectUnderTest.table("test_table");
+    objectUnderTest.payload("foo");
+    objectUnderTest.timestamp(1515585600000L);
+
+    final Message retval = objectUnderTest.build();
+
+    assertThat(retval).isNotNull();
+    assertThat(retval.getDatabase()).isEqualTo("test_db");
+    assertThat(retval.getTable()).isEqualTo("test_table");
+    assertThat(retval.getPayload()).isEqualTo("foo");
+    assertThat(retval.getTimestamp()).isEqualTo(1515585600000L);
```

Comment:
This test sort of includes all the tests that come after it - maybe if you remove all the assertions except the `isNotNull()`, that way this test just verifies that `.build()` "works" and the others verify that it does what it's meant to.